docs/workshops/interviews/technical/workshop.md

```diff
@@ -0,0 +1,262 @@
+
+### Get something working, then get it great
+
+Your next step is to come up with a solution. Some important guidelines:
+
+#### It's more important to come up with _a_ solution than the perfect solution.
+
+So get _something_ working, and then you can think about restructuring it if you want to improve it.
+
+For example, if you think you can either solve a problem with a `for` loop or with a few calls to `map` and `filter`, do whichever one you think is easiest.
+
+Once you've got that working, you can talk to your interviewer about improving it. e.g. after writing a solution with a `for` loop you may say "I think this would be cleaner if I used `filter`" - the interviewer may let you do that, may say "yeah, maybe, but don't worry about doing it, I'd rather talk about something else", or may ask you to talk through the difference without actually writing down the changes.
+
+One of the reasons for this is that if your interview runs out of time, if you've solved a problem an easier way, an interviewer can easily say "They solved the problem, and could talk about how they'd improve/finish it". If you try to solve a problem in a complicated way, and don't finish, the interviewer can only really say "They didn't solve the problem".
+
+#### It's really <ins>really</ins> important that you talk through what you're thinking, _particularly_ if you're stuck or confused.
+
+:::tip Remember
+An interviewer's goal is to see you at your best. They want you to succeed, which means if you need a little hint, they want to give it to you. But they can only do that if they know what you're stuck on.
+:::
+
+It can feel unnatural to talk through what you're thinking, _particularly_ when you're confused or stuck. But in a technical interview, it's _really_ important to do it anyway.
+
+Here's an example of how you might talk aloud for our example problem:
+
+> I need to find the split the string into words, and then I need to find the logest word...
+>
+> To split the string into words, I need to remove any punctuation, and then split it on whitespace.
+>
+> Let's start with that, and I can work out the longest word part after.
+>
+> Ok, so I know I can use String.replace to get rid of the punctuation, as long as I know what characters are punctuation... But I'm not sure how to work out which characters are punctuation.
+
+At which point your interviewer may say something like "Let's assume there's no punctuation" (because they're not interested in that part), or "assume you can call a function called stripPunctuation()" (because they know we use libraries for lots of things, and this isn't what they want to focucs on), or they may give you a hint, like "could we use a regular expression to find all of the characters that _aren't_ letters or whitespace?". But **if the interviewer doesn't know what you're thinking, they can't help** - and they want to help.
```

Comment:
I'm not sure this is the best place to put this, but we might want to cover something about googling. My sense is that most interviews these days are open book, so looking up specifics of regex would be ok.

It might also be worth stating that googling "find the longest word in a sentence in javascript" is probably not appropriate 😉